ast/src/main/java/com/graphicsfuzz/common/ast/type/QualifiedType.java

```diff
@@ -79,6 +80,34 @@ public void replaceQualifier(TypeQualifier oldQualifier, TypeQualifier newQualif
     qualifiers.add(newQualifier);
   }
 
+  /**
+   * Sets or replaces location qualifier.
+   *
+   * @param location Location to set in the location qualifier.
+   */
+  public void setLocationQualifier(int location) {
+    Optional existingLayoutQualifierSequence = Optional.empty();
```

Comment:
Change Optional to Optional<LayoutQualifierSequence>, because that's the type we expect to find in the Optional.  In general don't use "raw" types in Java (generic types with no parameter specified); they're only really there to allow interaction with legacy code.

Reply:
Assuming you meant Optional of LayoutQualifierSequence. It seems github eats things inside <>'s.